adlik_serving/framework/manager/boarding_functor.cc

```diff
@@ -285,8 +286,17 @@ struct ModelExecutor : ManagedNameVisitor {
   }
 
   void exec() {
-    if (auto action = next()) {
-      action->exec(store);
+    for (auto& name : names) {
+      actions.emplace_back(action(name));
+    }
+    while (true) {
+      auto action = next();
+      if (action) {
+        action->exec(store);
+        actions.erase(actions.begin());
```

Comment:
Be careful about actions (Vector) usage. Should use origin pop (pop-front) or other direct api. Pls add some testcases for sure.